diff --git a/contrib/src/aom/SHA512SUMS b/contrib/src/aom/SHA512SUMS
index c9f0c779a006118d290c13ba9e6d07c2676ac3b6..0f1fb40f9ce1cfb4b8b7ee3aeebc87f586e99f7f 100644
--- a/contrib/src/aom/SHA512SUMS
+++ b/contrib/src/aom/SHA512SUMS
@@ -1 +1 @@
-718c9cd41fe65c746a5e06b906b419fa919f65c999943101557ffd3edd1704179da69392083d877c0753327cb94b182664d34c7dedcd42696ec0c2bf55299248  libaom-3.13.1.tar.gz
+444f7abebcb568e13377d84b66e24cb61d3e81f6142b5763d8e65784a5da76821a1753827024e2c6bfdc09f737a3fe1c1805bbaff5108a15ca1400298c434ee4  libaom-3.13.2.tar.gz
diff --git a/contrib/src/aom/rules.mak b/contrib/src/aom/rules.mak
index 9f19f54374b1584be4fd605bd36958f85bf6e02c..9656047f3c55745902d638875955241a469c1594 100644
--- a/contrib/src/aom/rules.mak
+++ b/contrib/src/aom/rules.mak
@@ -1,5 +1,5 @@
 # aom
-AOM_VERSION := 3.13.1
+AOM_VERSION := 3.13.2
 AOM_URL := https://storage.googleapis.com/aom-releases/libaom-$(AOM_VERSION).tar.gz
 
 PKGS += aom
